shanoir-ng-datasets/src/main/java/org/shanoir/ng/examination/repository/ExaminationRepositoryImpl.java

```diff
@@ -72,9 +91,32 @@ private Pair<List<Examination>, Long> find(Iterable<Pair<Long, Long>> studyCente
 			nbPreParams++;
 			subjectNameIndex = nbPreParams;
 			queryEndStr +=  "and ex.subject.name is ?" + subjectNameIndex + " ";
-		} 
+		}
+		if (searchStr != null && !searchStr.isEmpty()) {
+			nbPreParams++;
+			searchStrIndex = nbPreParams;
+			if (searchField != null && !searchField.isEmpty()) {
+				if (searchField.equals("id")) {
+					queryEndStr += "and CAST(ex.id as String) LIKE CONCAT('%', ?" + searchStrIndex + ", '%') ";
+				} else if (searchField.equals("center.name")) {
+					queryEndStr += "and c.name LIKE CONCAT('%', ?" + searchStrIndex + ", '%') ";
+				} else if (searchField.equals("examinationDate")) {
+					queryEndStr += "and CAST(DATE_FORMAT(ex." + searchField + ", '%d/%m/%Y') as String) LIKE CONCAT('%', ?" + searchStrIndex + ", '%') ";
+				} else {
+					queryEndStr += "and ex." + searchField + " LIKE CONCAT('%', ?" + searchStrIndex + ", '%') ";
```

Comment:
Subject search does not seems to work for me here, only comment. As subjectName is not a direct attribute of 'examination' table, it can't work properly. I think you have to add a specific HQL part to look at subject name.